File: 1598. Crawler Log Folder.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        stack = []
        for i in logs:
            if i[-2] == '.':
                if len(i) >= 3 and i[-3] == '.':
                    if stack:
                        stack.pop()
                        
            else:
                stack.append(1)
        return len(stack)

File: 1598-crawler-log-folder/1598-crawler-log-folder.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int folder = 0;
        
        for (auto log: logs) {
            if (log == "../")
                folder--;
            folder = max(0, folder);
            if (log[0] != '.')
                folder++;
        }
        
        return folder;
    }
};